source/extensions/common/wasm/wasm.h

```diff
@@ -142,15 +142,19 @@ using WasmHandleSharedPtr = std::shared_ptr<WasmHandle>;
 
 class PluginHandle : public PluginHandleBase, public ThreadLocal::ThreadLocalObject {
 public:
-  explicit PluginHandle(const WasmHandleSharedPtr& wasm_handle, absl::string_view plugin_key)
-      : PluginHandleBase(std::static_pointer_cast<WasmHandleBase>(wasm_handle), plugin_key),
-        wasm_handle_(wasm_handle) {}
+  explicit PluginHandle(const WasmHandleSharedPtr& wasm_handle, const PluginSharedPtr& plugin)
+      : PluginHandleBase(std::static_pointer_cast<WasmHandleBase>(wasm_handle),
+                         std::static_pointer_cast<PluginBase>(plugin)),
+        wasm_handle_(wasm_handle),
+        config_id_(wasm_handle->wasm()->getRootContext(plugin, false)->id()) {}
 
   WasmSharedPtr& wasm() { return wasm_handle_->wasm(); }
   WasmHandleSharedPtr& wasmHandleForTest() { return wasm_handle_; }
+  uint32_t configId() { return config_id_; }
 
 private:
   WasmHandleSharedPtr wasm_handle_;
+  const uint32_t config_id_;
```

Comment:
should this be called `root_context_id_` in line with the arg name of Context and ContextBase's constructor?

Reply:
We pretty much moved away from the static `root_context_id`, which was updated in-place on config reloads, and we use a `context_id` that's bound to the configuration, so this name seemed more appropriate and I wanted to start the change somewhere.

But you're right, it might be confusing to have it done only here and not througout the codebase, so I'll move back to `root_context_id` for the time being to avoid confusion.